### PR TITLE
Performance Improvement for /dags Page

### PR DESCRIPTION
**Pull request by fritz-astronomer**
- add cache around DAGBag to prevent reparsing for each DAG on the /dags page

### CODE REVIEW

File: astronomer/starship/services/local_airflow_client.py [hunk 1]
```diff
@@ -104,7 +106,9 @@ def migrate(session: Session, table_name: str, dag_id: str):
     return rtn
 
 
-def receive_dag(data: list = []):
+def receive_dag(data: list = None):
+    if data is None:
+        data = []
```
Comment (fritz-astronomer) [inline astronomer/starship/services/local_airflow_client.py:111]: unrelated linter fix

File: astronomer/starship/main.py [hunk 2]
```diff
@@ -315,10 +315,11 @@ def dag_cutover_row_get(self, deployment: str, dag_id: str):
     def dag_cutover_row(
         self, deployment: str, dag_id: str, dest: str = "local", action: str = "init"
     ):
+        valid_actions = ["pause", "unpause", "migrate", "init"]
         if dest not in ["local", "astro"]:
             raise RuntimeError("dest must be 'local' or 'astro'")
-        if action not in ["pause", "unpause", "migrate", "init"]:
-            raise Exception("action must be 'pause', 'unpause', or 'migrate'")
+        if action not in valid_actions:
+            raise Exception(f"action must be in {valid_actions}")
```
Comment (frankcash) [inline astronomer/starship/main.py:322]: I think a better way to leverage this error message would be 

`            raise Exception(f"{action=} must be in {valid_actions=}")`

Which will show the current actiona nd allowed actions without making the person trying to run this package dive deep into the code base

Comment (fritz-astronomer) [reply astronomer/starship/main.py:322]: AWESOME! Haven't used that yet!

Comment (fritz-astronomer) [reply astronomer/starship/main.py:322]: Oh - hm, might not be available in all python we need to support due to `3.8`
https://docs.python.org/3/whatsnew/3.8.html#bpo-36817-whatsnew
